python/cudf/cudf/tests/test_string.py

```diff
@@ -2931,3 +2931,115 @@ def test_string_slice_with_mask():
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        [
+            """
+            {
+                "store":{
+                    "book":[
+                        {
+                            "category":"reference",
+                            "author":"Nigel Rees",
+                            "title":"Sayings of the Century",
+                            "price":8.95
+                        },
+                        {
+                            "category":"fiction",
+                            "author":"Evelyn Waugh",
+                            "title":"Sword of Honour",
+                            "price":12.99
+                        }
+                    ]
+                }
+            }
+            """
+        ],
+        [
+            """
+            {
+                "store":{
+                    "book":[
+                        {
+                            "category":"reference",
+                            "author":"Nigel Rees",
+                            "title":"Sayings of the Century",
+                            "price":8.95
+                        }
+                    ]
+                }
+            }
+            """,
+            """
+            {
+                "store":{
+                    "book":[
+                        {
+                            "category":"fiction",
+                            "author":"Evelyn Waugh",
+                            "title":"Sword of Honour",
+                            "price":12.99
+                        }
+                    ]
+                }
+            }
+            """,
+        ],
+    ],
+)
+@pytest.mark.parametrize(
+    "json_path", ["$.store", "$.store.book", "$.store.book[*].category", " "],
+)
+def test_string_get_json_object_n(data, json_path):
+    gs = cudf.Series(data)
+    gs.str.get_json_object(json_path)
+
+
+@pytest.mark.parametrize(
+    "json_path", ["$.store", "$.store.book", "$.store.book[*].category", " "]
+)
+def test_string_get_json_object_empty_json_strings(json_path):
+    gs = cudf.Series(
+        [
+            """{"": {
+        "": [
+        { "": "",
+          "": "",
+          "": "",
+        },
+        { "": "fiction",
+          "": "",
+          "title": "",
+        },
+        ]}}"""
+        ]
+    )
+
+    gs.str.get_json_object(json_path)
+
+
+@pytest.mark.parametrize("json_path", ["a", ".", "/.store"])
+def test_string_get_json_object_invalid_JSONPath(json_path):
+    gs = cudf.Series(
+        [
+            """{"store": {
+        "book": [
+        { "category": "reference",
+            "author": "Nigel Rees",
+            "title": "Sayings of the Century",
+            "price": 8.95
+        },
+        { "category": "fiction",
+            "author": "Evelyn Waugh",
+            "title": "Sword of Honour",
+            "price": 12.99
+        },
+        ]}}"""
+        ]
+    )
+
+    with pytest.raises(ValueError):
+        gs.str.get_json_object(json_path)
```

Comment:
We should add a test where we have more than one row in the `Series`. [Here's](https://stackoverflow.com/a/55948459) a nice example of how to generate random json data if it helps.

Reply:
Note, we don't have to generate random data within the test, but it could be useful for you instead of having to write json by hand.

Reply:
Actually`test_string_get_json_object` has two test cases : first has one row, second has two rows of data